loader/src/main.py

```diff
@@ -172,11 +174,26 @@ def process_event(msg):
             # references OR is referenced by other documents)
             logger.debug(
                 f"Resolving references {analysis.reference_paths}",
-                extra={"resource_id": resource_id},
+                extra={"batch_id": batch_id, "resource_id": resource_id},
+            )
+
+            # Redis distributed lock for reference binding
+            dlm = Redlock(
+                [{
+                    "host": REDIS_REFERENCES_HOST,
+                    "port": REDIS_REFERENCES_PORT,
+                    "db": REDIS_REFERENCES_DB
+                }, ]
             )
+            while True:
```

Comment:
So we'll try to take the lock until it's free?

Reply:
I think it should acquire the lock enventually... I could set a max_retries or a timeout or both... what do you think?